src/item_iterator.rs

```diff
@@ -186,6 +189,7 @@ pub struct PublicItemInner {
     /// The type info part, e.g. "(param_a: Type, param_b: OtherType)" for a
     /// `fn`.
     pub(crate) suffix: String,
+    pub tokens: tokens::TokenStream,
```

Comment:
Before I dive deeper I would like to stand a high-level choice of yours. Shouldn't we replace `prefix`, `path` and `suffix` with `tokens`? Rather than having them all.

Reply:
Ok, keeping `path` separate sounds reasonable now that you say so. So then maybe change the type of `prefix` and `suffix` from current `String` to `TokenStream`? So in  `impl Display for PublicItemInner` we do concatenate a token stream (the prefix), a string array (path), and another token stream (the suffix). I *think* it would be prettier to have also the `path` as a token stream, so that all components that represents a `PublicItem` is of the same type. But it's hard to make a final call without trying it out.

I don't like that the same data is represented in different ways in the same struct. I would like there to be a single source of truth. That generally results in a much simpler system. I hope that makes sense. My apologies in advance if I am completely misinterpreting your code.

I want to be clear that I love the work that you are doing, I hope I don't come across as negative here. I just want to give honest code review comments :)

Reply:
I do agree on having a single source of truth.
For clarity this is the struct I propose:
```rust
struct PublicItem {
    tokens: TokenStream,
    path: Vec<String>,
}
```
The major downside is that the path is encoded both in the `TokenStream` as `Token`s and in the path as `String`s. But I think this is easier to handle then separate `TokenStream`s as a public api.

And I am really grateful for your feedback, having two eyes on the same code always results in better code. And to be very honest I miss feedback from people who can code at approximately my level a bit in my day to day work.

Reply:
I agree that have a single token stream is an attractive proposition. So if you want to take a shot at using the struct you propose I am fine with that. So now my main concern is https://github.com/Enselic/public_items/pull/53#discussion_r841902208, i.e. that I would like to try to re-use the existing "if this then that" code rather than duplicating it. Which you talked about in:

> The rendering code need to support more input and I am thinking of a refactor into something like the D you made

But my advice would be to try to fit tokens in the existing "D-code" rather than replicating it another time. By no means trivial, but I feel like it would be the best approach.

Reply:
Great, I will work on some refactoring then and see how 'clean' I can get it. I tried a trait refactoring yesterday which did not really produce what I wanted, as I only need a `Render` trait or similar on like 4 types from the `rustdoc_types`. But I will try and see what I can do.